.github/workflows/CI.yml

```diff
@@ -0,0 +1,114 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: CI
+on:
+  push:
+    branches-ignore: [ 'dependabot/**' ]
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  single-commit:
+    name: Single Commit Pull Request
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check Single Commit
+        uses: actions/github-script@v5.0.0
+        with:
+          script: |
+            const commits = await github.rest.pulls.listCommits({
+              ...context.repo,
+              pull_number: context.issue.number,
+            });
+            core.info("Number of commits in this pull request: " + commits.data.length);
+            if (commits.data.length > 1) {
+              core.setFailed("If approved with two +1's, squash this pull request into one commit");
+            }
+
+  code-format:
+    name: TypeScript/Scala Formatting
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        shell: bash
+    env:
+      SBT: sbt -J-Xms1024m -J-Xmx5120m -J-XX:ReservedCodeCacheSize=512m -J-XX:MaxMetaspaceSize=1024m ++2.12.15
+
+    steps:
+      - uses: actions/checkout@v2.3.5
+      - name: Setup Node
+        uses: actions/setup-node@v1.4.6
+        with:
+          node-version: '16'
+      - name: Setup Java
+        uses: actions/setup-java@v2.3.1
+        with:
+          distribution: temurin
+          java-version: 11
+      - run: $SBT scalafmtCheck
+      - run: $SBT scalafmtSbtCheck
+      - run: yarn install
+      - run: yarn lint
+      - name: Run Rat Check
```

Comment:
@michael-hoke You can remove this line from this job since you do it in a separate one now